src/libs/PolicyUtils.ts

```diff
@@ -756,8 +756,11 @@ function getIntegrationLastSuccessfulDate(connection?: Connections[keyof Connect
     return (connection as ConnectionWithLastSyncData)?.lastSync?.successfulDate;
 }
 
-function getCurrentSageIntacctEntityName(policy?: Policy): string | undefined {
+function getCurrentSageIntacctEntityName(policy: Policy | undefined, translate: LocaleContextProps['translate']): string | undefined {
```

Comment:
I think it's really strange to be passing the `translate` function as a parameter. I'd prefer returning early in the method when there is no `currentEntityID` and having the calling function deal with the return value (maybe it should return `null`).

Reply:
Or my second suggestion would be to add a param to `getCurrentSageIntacctEntityName()` which is called `defaultNameIfNoEntity` and it would be called like:

```
getCurrentSageIntacctEntityName(policy, translate('workspace.common.topLevel'))
```

Yes, I think I like this approach the best.

Reply:
I have same thoughts with you initially, however after checking source code I see we also used "translate" as a function param in other places, so I think I'm fine with this current changes.

Reply:
> Or we can do somethings like this:
> 
> ```
> getCurrentSageIntacctEntityName(policy) ?? translate("topLevel")
> ```

I thought about that too but we would have to repeat it every time we use `getCurrentSageIntacctEntityName` function. It'd be better to handle this case inside the function

Reply:
To be clear, I don't think we should be doing that in the other places either. I think that translations are just a concern of the display and so they should only be handled in display components. Let's not keep spreading that concern out all over the app.